buildSrc/src/main/kotlin/plugins/KotlinDslPluginBundle.kt

```diff
@@ -0,0 +1,148 @@
+package plugins
+
+import com.gradle.publish.PluginBundleExtension
+import org.gradle.api.DomainObjectCollection
+import org.gradle.api.Named
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.BasePluginConvention
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.tasks.WriteProperties
+import org.gradle.api.tasks.testing.Test
+import org.gradle.language.jvm.tasks.ProcessResources
+import org.gradle.plugin.devel.GradlePluginDevelopmentExtension
+
+
+class KotlinDslPlugin(private val name: String) : Named {
+    override fun getName() = name
+    lateinit var displayName: String
+    lateinit var id: String
+    lateinit var implementationClass: String
+}
+
+
+open class KotlinDslPluginBundle : Plugin<Project> {
+
+    private
+    lateinit var kotlinDslPlugins: DomainObjectCollection<KotlinDslPlugin>
+
+    override fun apply(project: Project): Unit = project.run {
+
+        plugins.apply(KotlinDslModule::class.java)
+        plugins.apply("maven-publish")
+        plugins.apply("java-gradle-plugin")
+        plugins.apply("com.gradle.plugin-publish")
+
+        testsDependOnCustomInstallation()
+
+        kotlinDslPlugins = container(KotlinDslPlugin::class.java)
+        extensions.add("kotlinDslPlugins", kotlinDslPlugins)
+
+        configureGradlePluginDevelopmentPlugins()
+
+        workAroundTestKitWithPluginClassPathIssues()
+    }
+
+    private
+    fun Project.testsDependOnCustomInstallation() =
+        tasks.withType(Test::class.java).all {
```

Comment:
:x: `.all` is unnecessary, isn't it?

Reply:
I restored previous behavior, good catch.